### ISSUE TITLE
need to disable undo/redo in moanco-editor itself

### ISSUE THREAD

**Issue by jasonliu7040** [rnbwdev/rnbw#444]
now when you do undo/redo in monaco-editor focusing it, it is done with delay.
need to disable moanco-editor history itself, and implement it with the project hms we've built.

**Comment by edenvidal** [comment]
@atulbhatt-system32, does the mechanism to handle both code and events history exist? (we talked about it)

**Comment by jasonliu7040** [comment]
@atulbhatt-system32 
not exactly..
in a word, now it works differently when we do undo/redo focusing/not-focusing on code-view.

**Comment by jasonliu7040** [comment]
only for monaco

**Comment by edenvidal** [comment]
so, every coding session has its own 'history', right? so, now it is considered as one event, but it should be considered as a few since few things can happen while you're coding. for example - the creation, change, or removal of a single/multiple number of nodes.
@atulbhatt-system32 @jasonliu7040 it is worth aligning on this part pre-implementing it 🙌

**Comment by atulbhatt-system32** [comment]
@jasonliu7040 can you fill in with the implementation details.

**Comment by jasonliu7040** [comment]
so here's the current progress.
when you make any changes outside of code-view, it will be an "rnbw" event.
when you do typing in code-view, it will trigger debounced onChange handler. and debounced one will be an "rnwb" event.
when you do undo/redo, all of node-tree, stage-view, code-view are synced immediately.

so the issues is that if we do undo/redo in code-view, it doesn't work with our undo/redo mechanism since monaco-editor has its own mechanism.
need to disable that.

**Comment by jasonliu7040** [comment]
@edenvidal 
so let's suppose that I am typing in code-view.
I wrote 100 words in a fast speed. in the current version, it will be one event.
but what you want is to split that into several events?

**Comment by atulbhatt-system32** [comment]
@edenvidal During our discussion on history management, we decided to handle everything through code.
Every action changes the code (add/delete a node, etc).
We have implemented it in the same way. Every action ultimately changes the code in the codeview/editor. Our implementation is the same.

Also, then we also discussed that action/events like focusing nodes will also be a part of the history, and they also belong to that.
